File: Java/Eta/Applications/Shared/src/main/java/com/rtsdk/eta/shared/LoginRequestInfo.java
package com.rtsdk.eta.shared;

import com.rtsdk.eta.transport.Channel;
import com.rtsdk.eta.valueadd.domainrep.rdm.login.LoginMsgFactory;
import com.rtsdk.eta.valueadd.domainrep.rdm.login.LoginMsgType;
import com.rtsdk.eta.valueadd.domainrep.rdm.login.LoginRequest;

/** Login request information. */
public class LoginRequestInfo
{
    public LoginRequest loginRequest;
    Channel channel;
    boolean isInUse;

    public LoginRequestInfo()
    {
        loginRequest = (LoginRequest)LoginMsgFactory.createMsg();
        loginRequest.rdmMsgType(LoginMsgType.REQUEST);
    }

    public void clear()
    {
        loginRequest.clear();
        channel = null;
        isInUse = false;
    }
    
    public LoginRequest loginRequest()
    {
        return loginRequest;
    }
    
    public Channel channel()
    {
        return channel;
    }
    
    public boolean isInUse()
    {
        return isInUse;
    }
}

File: CSharp/Eta/Applications/Examples/EtaAppCommon/Rdm/LoginRequestInfo.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2022-2023 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.Transports;
using LSEG.Eta.ValueAdd.Rdm;

namespace LSEG.Eta.Example.Common
{
    public class LoginRequestInfo
    {
        public LoginRequest LoginRequest { get; private set; } = new LoginRequest();
        public IChannel? Channel { get; internal set; }

        public bool IsInUse { get; internal set; }

        public LoginRequestInfo()
        {
            Clear();
        }

        public void Clear()
        {
            LoginRequest.Clear();
            Channel = null;
            IsInUse = false;
        }
    }
}
